pkg/libvmi/vmi.go

```diff
@@ -105,16 +105,18 @@ func WithRng() Option {
 }
 
 // WithWatchdog adds a watchdog to the vmi devices.
-func WithWatchdog(action v1.WatchdogAction) Option {
+func WithWatchdog(action v1.WatchdogAction, arch string) Option {
 	return func(vmi *v1.VirtualMachineInstance) {
-		vmi.Spec.Domain.Devices.Watchdog = &v1.Watchdog{
+		watchdog := &v1.Watchdog{
 			Name: "watchdog",
-			WatchdogDevice: v1.WatchdogDevice{
-				I6300ESB: &v1.I6300ESBWatchdog{
-					Action: action,
-				},
-			},
 		}
+		if arch == "s390x" {
```

Comment:
This is problematic and unfortunate to see reaching main and backported.
Callers should not be required to know the architecture, it is a runtime (or sometime build) thing.

I wont block on this now, but it needs to be taken care of as a follow up.

Reply:
I will work on this in the next PR